main.go

```diff
@@ -113,7 +115,9 @@ func main() {
 	log.Printf("Listening on: %v\n", h.Addrs())
 
 	// create a gossipsub node and subscribe to the topic
-	psOpts := pubsubOptions()
+	// dice will never be 0, so numMaliciousFlag = 0 will yield a 100% honest network
+	dice := rand.Intn(99) + 1
+	psOpts := pubsubOptions(dice <= *numMaliciousFlag && nodeId != 0)
```

Comment:
Could you just have a flag indicating whether to ignore ineeds or not and toss the dice in network_graph.py instead?

Reply:
makes sense. adding it